### ISSUE TITLE
B-Table "items" property -> destroy data during pagination

### ISSUE THREAD

**Issue by susanne99** [bootstrap-vue/bootstrap-vue-next#1050]
it is very difficult to explain and to reproduce.
my app consists of a lot of b-tables.
the "items" prop of all these tables will be feed by a computed property (only a getter !)
the computed property perpares the sourcedata for the table

**in the most cases, the sourcedata will be available via prop - then it is all okay, the table works as expected, also the pagination.
but in the other cases, where the source data are come from local data (not via prop) of the component, then the pagination does not work at all. that means, that the "items" prop of b-table destroys the source data, also when it comes from a computed property which should be readonly, because i use it only as getter !**

I know it sounds funny but that's how it is. I've been looking and comparing for a few days now I can't find the error, maybe it's also in the area of ​​vuejs 3

**Comment by susanne99** [comment]
here a code snippet of my app: important is the prop: "items" with the computed prop: "PreparedReceiptList"

```
 <b-table
                                show-empty small striped hover bordered responsive
                                select-mode="single"
                                stacked="md"
                                head-variant="light"
                                table-variant="secondary"
                                empty-text="Es sind keine Daten vorhanden"
                                empty-filtered-text="Es sind keine Daten vorhanden"
                                :busy="data_loading"
                                :items="PreparedReceiptList"
                                :fields="grid_fields"
                                :current-page="currentPage"
                                :per-page="perPage"
                                :filter="filter"
                                :filterIncludedFields="filterOn"
                                v-model:sort-by="sortBy"
                                v-model:sort-desc="sortDesc"
                                :sort-direction="sortDirection"
                                @filtered="onFiltered_Receipt_Grid">
```

variant 1 with component prop (diagnosislist) (example from another working component):
```
 props: {
        diagnosislist: { type: Array as PropType<any[]>, default: () => [] },
    },
```

variant 2 with local data (receiptlist)
```
data() {
        return {
            receiptlist: [] as any[],
            data_loading: false,

            grid_fields: [
                { key: 'id', label: 'ID', sortable: false, class: 'pms-hide' },
                { key: 'invoicetype', label: 'Belegtyp', sortable: false },
                { key: 'invoicenumber', label: 'Belegnummer', sortable: false },
                { key: 'invoicedate_txt', label: 'Belegdatum', sortable: false },
                { key: 'totalamount_txt', label: 'Gesamtbetrag [Brutto]', sortable: false, class: 'text-end' },
                { key: 'options', label: 'Optionen', class: 'pms-options-column' }
            ],
            totalRows: 1,                // computed property
            currentPage: 1,
            perPage: 10,
            pageOptions: [5, 10, 20],
            sortBy: 'id',
            sortDesc: true,                 // sort id descending
            sortDirection: 'asc',
            filter: '',
            filterOn: filterfieldlist,
        }
    },

computed: {
        PreparedReceiptList(): any[] {
            console.log('here am i');

            if (this.receiptlist == null) { return []; }

            let receiptlist: any[] =  Helper.DeepCopy(this.receiptlist);

            receiptlist.forEach((receipt: any) => {
                receipt.paid = !!receipt.paid;                                                              // convert 1,0 to true, false
                receipt.failure = !!receipt.failure;                                                        // convert 1,0 to true, false
                receipt.cancellation = !!receipt.cancellation;                                              // convert 1,0 to true, false    
            })

            return receiptlist;
        },
    },

```
```
 CashRegisterService.getReceiptListByDate(this.datefrom, this.dateto, this.invoicetype)
                .then((response: any) => {
                    this.data_loading = false;
                    this.receiptlist = Helper.Json2Array(response.data.receiptlist);
                })
```



this example is used in many locations in my app.
A list (this.receiptlist) will be either filled by a component prop, or like in this case with a local method which loads the data and stores in a local data variable.
when i change the currentpage, then the items property value (computed property value: PreparedReceiptList )changed also, that means, lets assume you have an receiptlist array of 30 member, after the page change you have 20 member in it, although no new data was loaded. again, when i show the result of PreparedReceiptList, the the list consists only 20 members. i have watched this with console.log, {{ PreparedReceiptList.length }} and vue dev tools in chrome. 
i placed also a console.log in the PreparedReceiptList function, but it is not called after currentpage change - that is right
but something changed the result of PreparedReceiptList without calling it - i know it sounds strange - maybe there is a vuejs 3 bug anywhere

what I noticed, but not 100% confirm, is that when the receipt list comes from a component property, this error isn't there

I tried to reproduce the error in stackblitz, but that was too complicated for me

**Comment by susanne99** [comment]
solution: 
#1047

**Comment by RuBaWa** [comment]
Please try current version. Bug with pagination and mutating item prop is fixed. Maybe this issue is also gone

**Comment by susanne99** [comment]
yes, fixed in 0.8.6
